src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -302,8 +309,20 @@ class MariaDBManager implements IQueryService {
         int personID = rs.get("responsible_person_id");
         Person pers = getPersonNameByID(personID)
 
-        if(currID == locID) {
-          currLoc = new Location(name: name, responsiblePerson: pers.getFirstName()+" "+pers.getLastName(), responsibleEmail: pers.getEMail(), address: address, status: status, arrivalDate: arrivalDate, forwardDate: forwardedDate);
+        if(currID == locID ) {
+           // Compare if current location is the newest entry
+            if (currDate && currDate.before(arrivalDate)) {
+              // Set the current location to the newer one
+              log.info("Newer location entry found!")
+              pastLocs.add(new Location(name: currLoc.name, responsiblePerson: currLoc.name,
+                      responsibleEmail: currLoc.responsibleEmail, address: currLoc.address,
+                      status: currLoc.status, arrivalDate: currDate))
+              currDate = arrivalDate
+              currLoc = new Location(name: name, responsiblePerson: pers.getFirstName()+" "+pers.getLastName(), responsibleEmail: pers.getEMail(), address: address, status: status, arrivalDate: arrivalDate, forwardDate: forwardedDate);
+            } else {
```

Comment:
I do not understand this case. Here you expect the `currDate` to be null or to be after the `arrivalDate`. What happens then and why? Where is the newly created location `currLoc` used here? Shouldn't there also be a check for the arrival date here? Might this be related to #13 ?

Reply:
If the location that is currently evaluated has a newer arrival date then the one last set as current location, we need to set this location to the actual current location.

Reply:
Apologies, I know the code is not easy to read, but that is the current code base. Would need a shitload of refactoring otherwise.

Reply:
yes but the case I do not understand is when this the other one.

Reply:
So when the location is not newer then no location is added but a location is modified here. That is what confuses me.

Reply:
if the location is older than it is added in the history of past locations.

Reply:
```suggestion
            } else {
                if  (currLoc) {
                    log.warn("The sample was changed by someone else in the meantime. Please set a correct date")
                }
```
Or similar

Reply:
0fd6d34